src/main/java/com/hazelcast/gcp/GcpDiscoveryStrategyFactory.java

```diff
@@ -50,4 +50,47 @@ public DiscoveryStrategy newDiscoveryStrategy(DiscoveryNode discoveryNode, ILogg
         }
         return result;
     }
+
+    @Override
+    public boolean isAutoDetectionApplicable() {
+        return googleInternalDnsConfigured() && metadataFlavorGoogle();
+    }
+
+
+    private String readFileContents(String fileName) {
+        InputStream is = null;
+        try {
+            File file = new File(fileName);
+            byte[] data = new byte[(int) file.length()];
+            is = new FileInputStream(file);
+            is.read(data);
+            return new String(data, StandardCharsets.UTF_8);
+        } catch (IOException e) {
+            throw new RuntimeException("Could not get " + fileName, e);
+        } finally {
+            IOUtil.closeResource(is);
+        }
+    }
+
+    private boolean googleInternalDnsConfigured() {
```

Comment:
Please reorder the methods. They should follow [Stepdown rule](https://dzone.com/articles/the-stepdown-rule)

Reply:
I did it following the rule.